### PR TITLE
Componetize all the things

### PR DESCRIPTION
**Pull request by PorgJuice**
Massive changes to the structure of the app, my friends.  I basically just went through and did the tedious work of shoving markup into functional stateless components.  A bit of redundancy was reduced, but the most important thing is that the app is _easily_ testable now.  I also moved the list of "Pairing Techs" into App.js as an array, so adding or removing a tech from the list is as easy as editing the array.

There are a couple of problems.  The biggest one is that the "Other" text field in the Add modal doesn't work anymore.  It doesn't seem to have any state passed to it.  Easy to fix, but I wanted input on that.  The second problem is that I couldn't test how the users look when listed as I didn't have the backend running.  Hope this helps!

### CODE REVIEW

**Comment by JacksonBates** [comment]
> The biggest one is that the "Other" text field in the Add modal doesn't work anymore. 

This is an outstanding issue which we've been meaning to get around to.

I think the dev environment is still wired up to the live db, so you should be able to see the changes on users for real if your refresh your local version - but maybe we already locked that down...I've lost track :)

I'll run it from my end with everything to check it.

File: src/components/ModalForm.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import ModalSelectionList from './ModalSelectionList';
+
+const ModalForm = props => {
+  return (
+    <form onSubmit={props.handleSubmit}>
+        <AddModalUsernameInput username={props.username} handleChange={props.handleChange} />
+        <AddModalAvailableTimeInput availableTime={props.availableTime} handleChange={props.handleChange} />
+        <ModalSelectionList modalSelections={props.modalSelections}/>
+        <AddModalTextInput name="setup" label="Other" value="" handleChange={props.handleChange} />
+        <AddModalTextInput name="interests" label="Interests" value={props.interests} handleChange={props.handleChange} />
```
Comment (JacksonBates) [inline src/components/ModalForm.js:14]: This is missing the submit button

**Review by JacksonBates — CHANGES_REQUESTED** [review]
Only thing I've noticed is the missing submit button from the Modal add form.

File: src/components/ModalForm.js (hunk 1 above)
Comment (PorgJuice) [reply src/components/ModalForm.js:14]: Added, and the person responsible for omitting the button has been sacked.

Comment (JacksonBates) [reply src/components/ModalForm.js:14]: Have you pushed that to this branch? I'm not seeing it

Comment (PorgJuice) [reply src/components/ModalForm.js:14]: It's actually been added this time, and the person responsible for omitting the button a second time has also been sacked.

**Review by JacksonBates — APPROVED** [review]

**Pull request merged**